engine/apps/alerts/tasks/notify_user.py

```diff
@@ -186,6 +187,17 @@ def notify_user_task(
                         notification_channel=notification_policy.notify_by,
                     )
         if log_record:  # log_record is None if user notification policy step is unspecified
+            # if this is the first notification step, and user hasn't been notified for this alert group - update metric
+            if (
+                log_record.type == UserNotificationPolicyLogRecord.TYPE_PERSONAL_NOTIFICATION_TRIGGERED
+                and previous_notification_policy_pk is None
+                and not user.personal_log_records.filter(
+                    type=UserNotificationPolicyLogRecord.TYPE_PERSONAL_NOTIFICATION_TRIGGERED,
+                    alert_group_id=alert_group_pk,
+                ).exists()
+            ):
+                metrics_update_user_cache(user)
```

Comment:
`MetricsCacheManager` was added for preparing data for caching, e.g. for bulk updates. Since this part was reworked the cache manager makes less sense. We need to check if we can get rid of this.

In the case with the new metric we only update the counter and don't have data to prepare.